plugins/main/public/components/endpoints-summary/register-agent/containers/register-agent/register-agent.tsx

```diff
@@ -0,0 +1,268 @@
+import { Steps } from '../steps/steps';
+import { InputForm } from '../../../../common/form';
+import {
+  getGroups,
+  getMasterRemoteConfiguration,
+} from '../../services/register-agent-services';
+import { useForm } from '../../../../common/form/hooks';
+import { FormConfiguration } from '../../../../common/form/types';
+import { useSelector } from 'react-redux';
+import {
+  withErrorBoundary,
+  withReduxProvider,
+  withGlobalBreadcrumb,
+  withUserAuthorizationPrompt,
+} from '../../../../common/hocs';
+import GroupInput from '../../components/group-input/group-input';
+import { OsCard } from '../../components/os-selector/os-card/os-card';
+import {
+  validateServerAddress,
+  validateAgentName,
+} from '../../utils/validations';
+import { compose } from 'redux';
+import { endpointSumary } from '../../../../../utils/applications';
+import { getCore } from '../../../../../kibana-services';
+import { getErrorOrchestrator } from '../../../../../react-services/common-services';
+
+export const RegisterAgent = compose(
+  withErrorBoundary,
+  withReduxProvider,
+  withGlobalBreadcrumb([
+    { text: endpointSumary.title, href: `#${endpointSumary.redirectTo()}` },
+    { text: 'Deploy new agent' },
+  ]),
+  withUserAuthorizationPrompt([
+    [{ action: 'agent:create', resource: '*:*:*' }],
+  ]),
+)(() => {
+  const configuration = useSelector(
+    (state: { appConfig: { data: any } }) => state.appConfig.data,
+  );
+  const [wazuhVersion, setWazuhVersion] = useState('');
+  const [haveUdpProtocol, setHaveUdpProtocol] = useState<boolean | null>(false);
+  const [loading, setLoading] = useState(false);
+  const [wazuhPassword, setWazuhPassword] = useState('');
+  const [groups, setGroups] = useState([]);
+  const [needsPassword, setNeedsPassword] = useState<boolean>(false);
+
+  const initialFields: FormConfiguration = {
+    operatingSystemSelection: {
+      type: 'custom',
+      initialValue: '',
+      component: props => {
+        return <OsCard {...props} />;
+      },
+      options: {
+        groups,
+      },
+    },
+    serverAddress: {
+      type: 'text',
+      initialValue: configuration['enrollment.dns'] || '',
+      validate: validateServerAddress,
+    },
+    agentName: {
+      type: 'text',
+      initialValue: '',
+      validate: validateAgentName,
+    },
+
+    agentGroups: {
+      type: 'custom',
+      initialValue: [],
+      component: props => {
+        return <GroupInput {...props} />;
+      },
+      options: {
+        groups,
+      },
+    },
+  };
+
+  const form = useForm(initialFields);
+
+  const getRemoteConfig = async () => {
+    const remoteConfig = await getMasterRemoteConfiguration();
+    if (remoteConfig) {
+      setHaveUdpProtocol(remoteConfig.isUdp);
+    }
+  };
+
+  const getAuthInfo = async () => {
+    try {
+      const result = await WzRequest.apiReq(
+        'GET',
+        '/agents/000/config/auth/auth',
+        {},
+      );
+      return (result.data || {}).data || {};
```

Comment:
Maybe this could be `result?.data?.data || {};`

Reply:
Done